src/libs/markAllPolicyReportsAsRead.ts

```diff
@@ -0,0 +1,32 @@
+// eslint-disable-next-line you-dont-need-lodash-underscore/each
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Report} from '@src/types/onyx';
+import * as ReportActionFile from './actions/Report';
+
+export default function markAllPolicyReportsAsRead(policyID: string) {
+    const connectionID = Onyx.connect({
+        key: ONYXKEYS.COLLECTION.REPORT,
+        waitForCollectionCallback: true,
+        callback: (allReports) => {
+            if (!allReports) {
+                return;
+            }
+
+            let delay = 0;
+            Object.keys(allReports).forEach((key: string) => {
+                const report: Report | null | undefined = allReports[key];
+                if (report?.policyID !== policyID) {
+                    return;
+                }
+
+                setTimeout(() => {
```

Comment:
Can we check if the report is unread with `ReportUtils::isUnread` before making the api call?

Reply:
Sounds reasonable to me